app/controllers/users/reactivate_profile_controller.rb

```diff
@@ -19,7 +19,12 @@ def create
     def build_reactivate_profile_form
       ReactivateProfileForm.new(
         current_user,
-        params[:reactivate_profile_form].permit(:recovery_code, :password)
+        params[:reactivate_profile_form].
+        permit({:recovery_code => []}, :password).
+        tap do |obj|
+          code = obj[:recovery_code]
+          obj[:recovery_code] = code.join(' ') unless obj[:recovery_code].nil?
```

Comment:
So, my question wasn't whether or not it would work, it was if you knew why rails accepted the new hash syntax if the parameter was _after_ the `:password`, but not before.